packages/renderer-protocol/inverseRpc/services/friendRequestService.ts

```diff
@@ -114,14 +84,37 @@ export function registerFriendRequestKernelService(port: RpcServerPort<RendererP
 type FriendshipError = { message: { $case: 'error'; error: FriendshipErrorCode } }
 
 /**
- * Build friend requests error message to send to renderer
- * @param error - an int representing an error code
+ * Build friend requests error message to send to renderer.
+ * @param error - an int representing an error code.
  */
-function buildFriendRequestsError(error?: FriendshipErrorCode): FriendshipError {
+function buildErrorResponse(error?: FriendshipErrorCode): FriendshipError {
   return {
     message: {
       $case: 'error' as const,
       error: error ?? FriendshipErrorCode.FEC_UNKNOWN
     }
   }
 }
+
+/**
+ * Build friend requests success message to send to renderer.
+ * @param reply - a FriendRequestReplyOk kind of type.
+ */
+function wrapReply<T>(reply: T) {
+  return { message: { $case: 'reply' as const, reply: reply as NonNullable<T> } }
+}
+
+/**
+ * Build friend requests message to send to renderer.
+ * If the friendRequest object is truthy, the function returns the result of calling `wrapReply` on the `friendRequest` object.
+ * If the friendRequest object is falsy, the function returns the result of calling `buildErrorResponse` with the `error` value.
+ * @param friendRequest - it can represent any kind of FriendRequestReplyOk type.
+ * @param error - an int representing an error code.
+ */
+function buildResponse<T>(friendRequest: T, error?: FriendshipErrorCode) {
```

Comment:
From the use this is given, why not just sending the object with both the reply and error instead (using a single parameter)?

Reply:
I thought about it and I ended up having sth like this because TypeScript was reporting non-existent properties in the generic function:
```ts
type ExtendedType<T, U> = T & { reply?: U; error?: FriendshipErrorCode }

function buildResponse<T, U>(friendRequest: ExtendedType<T, U>) {
  if (friendRequest.reply) {
    return wrapReply(friendRequest.reply)
  } else {
    return buildErrorResponse(friendRequest.error)
  }
}
```

I didn't particularly like the solution tbh. But yes, I think we can definitely go in that direction. Give me a thumbs up and I'll commit that solution.

Reply:
Let me check that a sec!